packages/KDateTimePicker/KDateTimePicker.vue

```diff
@@ -0,0 +1,287 @@
+    timeframeLength: 60 * 15
+  },
+  {
+    section: 'Last',
+    timeframeText: 'Hour',
+    timeframeLength: 60 * 60 * 1
+  },
+  {
+    section: 'Last',
+    timeframeText: '3 hours',
+    timeframeLength: 60 * 60 * 3
+  },
+  {
+    section: 'Last',
+    timeframeText: '6 hours',
+    timeframeLength: 60 * 60 * 6
+  },
+  {
+    section: 'Last',
+    timeframeText: '12 hours',
+    timeframeLength: 60 * 60 * 12
+  },
+  {
+    section: 'Last',
+    timeframeText: '24 hours',
+    timeframeLength: 60 * 60 * 24
+  },
+  {
+    section: 'Last',
+    timeframeText: '7 days',
+    timeframeLength: 60 * 60 * 24 * 7
+  },
+  {
+    section: 'Last',
+    timeframeText: '30 days',
+    timeframeLength: 60 * 60 * 24 * 30
+  },
+  {
+    section: 'Last',
+    timeframeText: '90 days',
+    timeframeLength: 60 * 60 * 24 * 90
+  },
+  {
+    section: 'Previous',
+    timeframeText: 'Week',
+    timeframeLength: 60 * 60 * 24 * 90
+  },
+  {
+    section: 'Previous',
+    timeframeText: 'Month',
+    timeframeLength: 60 * 60 * 24 * 90
+  },
+  {
+    section: 'Previous',
+    timeframeText: 'Quarter',
+    timeframeLength: 60 * 60 * 24 * 90
+  }
+]
+
+export default {
+  name: 'KTimePicker',
+  components: {
+    KButton,
+    KPop,
+    KSegmentedControl,
+    DatePicker: () => import('v-calendar/lib/components/date-picker.umd')
+  },
+  props: {
+    value: {
+      type: Object,
+      required: true
+    }
+  },
+
+  data () {
+    return {
+      allowedTimePeriods,
+      mode: this.mode || 'custom',
+      selectimedTimeRange: this.selectimedTimeRange,
+      selectedTimeframe: this.value || this.allowedTimePeriods[0],
+      modelConfig: {
+        type: 'number'
+      }
+    }
+  },
+
+  computed: {
+    showCalendar () {
+      return (this.mode === 'custom')
+    }
+  },
+
+  watch: {
+    selectimedTimeRange: {
+      handler (newValue, oldValue) {
+        // console.warn('>>>> selectimedTimeRange watcher <<<')
+        // console.warn(newValue)
+      },
+      immediate: true
+    }
+  },
+
+  methods: {
+    /**
+     * TODO: Given a relative time frame, determine `start` and `end` timestamps in UTC
+     * https://bloop.ai/search/how-to-use-date-fns-library-round-to-nearest-minutes-in-javascript
+     * @param {*} timeframe
+     */
+    changeTimeframe (timeframe) {
+      this.selectedTimeframe = timeframe
+    },
+    clearSelection () {
+      this.selectimedTimeRange = ''
+      this.selectedTimeframe = this.allowedTimePeriods[0]
+    },
+    submitTimeFrame () {
+      console.warn('>>> submitTimeFrame')
+      let start = 0
+      let end = 0
+
+      // If calendar currently shown, send start/end values
+      if (this.showCalendar) {
+        start = Math.floor(this.selectimedTimeRange.start / 1000)
+        end = Math.floor(this.selectimedTimeRange.end / 1000)
+      } else {
+        end = getUnixTime(roundToNearestMinutes(Date.now()))
+        start = end - this.selectedTimeframe.timeframeLength
```

Comment:
this.$emit('changed', selectedTimeframe)

Reply:
What is `Timeframes.ts` ?

Reply:
It's where we stored the display text / actual timestamp value mapping:
https://github.com/Kong/khcp-ui/pull/2479/files#diff-017f16348af343f9855ca2f6c0e29e2bffed3e08cb17b9052dcec6af3a83d733

Thanks to @filipgutica work in that PR, we now have extended functionality that also dynamically determines `start` and `end` values (to support the newly requested **Previous** and **Current** time frames).

Reply:
So keep in mind you don't have access to that here in Kongponents, so is the thought that `Timeframes.ts` becomes something that lives inside Kongponents, or..? 

If yes, it wouldn't be something you all (i.e. Analytics team) could be changing all the time on the fly. Once it's in Kongponents, it's more of an uphill fight to make changes since this repo is open-source.

If it's something that is really specific to Konnect, I'd recommend allowing whatever it configures to be passed in via a prop or something.